final_assignment/Alexandru Tetelea/warhouse.py

```diff
@@ -0,0 +1,207 @@
+
+
+def ShutDown(type="soft"):
+    message = 'ShutDown'
+    timeout = 5
+    logging.info("Restart! ")
+    if str(sys.platform).__contains__("linux"):
+        try:
+            if type == "soft":
+                os.system('/sbin/shutdown now')
+            elif type == "hard":
+                os.system('/sbin/poweroff')
+        except:
+            logging.warn("Posibil sa nu avti drepturile necesare!")
+    elif str(sys.platform).__contains__("win"):
+        try:
+            import win32api
+            if type == "soft":
+                win32api.InitiateSystemShutdown(None, message, timeout, 0, 0)
+            elif type == "hard":
+                win32api.InitiateSystemShutdown(None, message, timeout, 1, 0)
+        except:
+            logging.warn("Posibil sa nu aveti drepturile necesare!")
+
+
+def delete(type="force", path=os.getcwd()):
+    logging.info("Delete File")
+
+    if os.path.exists(path):
+        try:
+            if type=="soft":
+                shutil.rmtree(path)
+                logging.info("Sters cu succes!: " + path)
+            elif type=="hard":
+                os.unlink(path)
+                os.rmdir(path)
+        except:
+
+            logging.warn("Nu s-a putut sterge!: " + path)
+    else:
+        logging.warn("Inexistent!: " + path)
+
+
+def config(configData):
+    logging.info("Config System")
+
+    os.system(configData["hostname"])
+    for key, value in configData["users"].iteritems():
+        createUser(value["full_name"], key, value["password"])
+
+
+def createUser(name, username, password):
+    logging.info("Create user")
+    encPass = crypt.crypt(password, "22")
+    return os.system(
+        "useradd -p " + encPass + " -s " + "/bin/bash " + "-d " + "/home/" + username + " -m " + " -c \"" + name + "\" " + username)
+
+
+def main(file):
+    with open(file) as data_file:
+        data = json.load(data_file)
+        fail = 0
+        for need in data["before_install"]:
+            Download(need["download"]["source"], need["download"]["destination"])
+        if "install" in dict(data).keys():
+            for task in data["install"]:
+                attempts = task["run_script"]["attempts"]
+                allowed_exit_codes = task["run_script"]["check_exit_code"]
+                if allowed_exit_codes.lower() == "true":
+                    allowed_exit_codes = bool("true")
+                elif allowed_exit_codes.lower() == "false":
+                    allowed_exit_codes = bool("")
+                command = task["run_script"]["command"]
+                cwd = task["run_script"]["cwd"]
+                env_variables = task["run_script"]["env_variables"]
+                retry_interval = task["run_script"]["retry_interval"]
+                shell = task["run_script"]["shell"]
+                fail = install_program(attempts, allowed_exit_codes, command, cwd, env_variables, retry_interval, shell)
+            if "after_install" in dict(data).keys():

```

Comment:
Am aplicat asta, pentru a informa ide in mod direct ce tip va fi acolo. Da, ar fi trebuit sa-l scot dupa.